test/integration/oauth_basicauth_test.go

```diff
@@ -195,35 +223,17 @@ func TestOAuthBasicAuthPassword(t *testing.T) {
 		}
 		if req.Header.Get("Authorization") != expectedAuthHeader {
 			w.WriteHeader(http.StatusUnauthorized)
-			t.Fatalf("Unexpected auth header: %s", req.Header.Get("Authorization"))
+			return
```

Comment:
This is part of the first of the three patches. I added functionality to do a negative test: proof that attempting to log in with the wrong password would be blocked. In order to do that, we needed to have the auth server return `http.StatusUnauthorized` rather than failing the test.

See below.